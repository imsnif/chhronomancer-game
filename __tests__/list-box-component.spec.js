import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('ListBox (with title) => renders correctly', () => {
  const ListBox = require('../components/list-box').default
  const tree = renderer.create(
    <ListBox title='foo' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ListBox (with title and subtitle) => renders correctly', () => {
  const ListBox = require('../components/list-box').default
  const tree = renderer.create(
    <ListBox title='foo' subtitle='bar' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ListBox (with title and picture) => renders correctly', () => {
  const ListBox = require('../components/list-box').default
  const tree = renderer.create(
    <ListBox title='foo' image={{uri: 'http://foo.bar/baz'}} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ListBox (with title, subtitle and picture) => renders correctly', () => {
  const ListBox = require('../components/list-box').default
  const tree = renderer.create(
    <ListBox title='foo' subtitle='bar' image={{uri: 'http://foo.bar/baz'}} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
