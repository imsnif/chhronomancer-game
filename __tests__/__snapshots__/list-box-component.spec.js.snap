// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBox (with title and picture) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://foo.bar/baz",
        }
      }
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "black",
          "borderWidth": 1,
          "flex": 0,
          "height": 28,
          "margin": 1,
          "resizeMode": "stretch",
          "width": 26,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "height": "100%",
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </Image>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`ListBox (with title and subtitle) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "paddingLeft": 5,
          "textAlign": "left",
        }
      }
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 3,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      bar
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`ListBox (with title) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`ListBox (with title, subtitle and picture) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://foo.bar/baz",
        }
      }
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "black",
          "borderWidth": 1,
          "flex": 0,
          "height": 28,
          "margin": 1,
          "resizeMode": "stretch",
          "width": 26,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "height": "100%",
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </Image>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "paddingLeft": 5,
          "textAlign": "left",
        }
      }
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 3,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      bar
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  />
</View>
`;
